models/articles.models.js

```diff
@@ -6,4 +6,22 @@ exports.selectArticleById = (article_id) => {
         if(rows.length===0){return Promise.reject({ message: "article does not exist", status: 404})}
         return rows[0];
       });
+  };
+
+  exports.selectArticles = () => {
+    let queryStr = 'SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url FROM articles'
+
+    queryStr += ` ORDER BY created_at DESC;`
+    return db
+      .query(queryStr)
+      .then(({rows}) => {
+        rows.forEach((article)=>{
+          let count = 0
+          let articleId = article.article_id
+          const comments = require('../db/data/test-data/comments')
+          comments.forEach((comment)=>{if(comment.article_id === articleId){count++}})
+            article.comment_count = count
+        })
```

Comment:
Your feeling is correct that this is not really the way you should do it. 

You should use an aggregate function in the query to create this extra column. Have a look at COUNT 😄